reV/econ/econ.py

```diff
@@ -153,6 +159,228 @@ def __init__(self, points_control, cf_file, cf_year, site_data=None,
+                    else:
+                        # this is the scenario where the project points
+                        # is a subset of the meta data like for
+                        # test problems.
+                        pc = PointsControl(
+                            pp, sites_per_split=sites_per_worker)
+
+        except FileNotFoundError as ex:
+            msg = ('{} is invalid PointsControl can not be adjusted!\n{}'
+                   .format(cf_file, ex))
+            raise ProjectControlFileError(msg)
+
+        return pc
+
+    @staticmethod
+    def get_pc(points, points_range, sam_files, tech, sites_per_worker=None,
+               cf_file=None, append=False):
+        """
+        Get a PointsControl instance.
+
+        Parameters
+        ----------
+        points : slice | list | str | reV.config.project_points.PointsControl
+            Slice specifying project points, or string pointing to a project
+            points csv, or a fully instantiated PointsControl object.
+        points_range : list | None
+            Optional two-entry list specifying the index range of the sites to
+            analyze. To be taken from the reV.config.PointsControl.split_range
+            property.
+        sam_files : dict | str | list | SAMConfig
+            SAM input configuration ID(s) and file path(s). Keys are the SAM
+            config ID(s), top level value is the SAM path. Can also be a single
+            config file str. If it's a list, it is mapped to the sorted list
+            of unique configs requested by points csv. Can also be a
+            pre loaded SAMConfig object.
+        tech : str
+            SAM technology to analyze (pvwattsv7, windpower, tcsmoltensalt,
+            solarwaterheat, troughphysicalheat, lineardirectsteam)
+            The string should be lower-cased with spaces and _ removed.
+        sites_per_worker : int
+            Number of sites to run in series on a worker. None defaults to the
+            resource file chunk size.
+        cf_file : str
+            reV generation capacity factor output file with path.
+        append : bool
+            Flag to append econ datasets to source cf_file. This has priority
+            over the fout and dirout inputs.
+
+        Returns
+        -------
+        pc : reV.config.project_points.PointsControl
+            PointsControl object instance.
+        """
+        pc = Gen.get_pc(points, points_range, sam_files, tech,
+                        sites_per_worker=sites_per_worker,
+                        res_file=cf_file)
+
+        if append and cf_file is not None:
+            try:
+                pc = Econ._econ_append_pc(pc.project_points, cf_file,
+                                          sites_per_worker=sites_per_worker)
+            except Exception:
+                msg = ("Failed to adjust ProjectControl for Econ in append "
+                       "mode!")
+                logger.exception(msg)
+                raise
```

Comment:
Shouldn't you raise the exception? Otherwise the stderr will just say no active exception to raise?

```
except Exception as e:
    ...
    raise e